lib/providers/notification/notification_provider.dart

```diff
@@ -5,22 +5,57 @@ import 'package:Medito/providers/providers.dart';
 import 'package:Medito/services/notifications/notifications_service.dart';
 import 'package:firebase_messaging/firebase_messaging.dart';
 import 'package:flutter/material.dart';
-import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:go_router/go_router.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+part 'notification_provider.g.dart';
 
-final checkNotificationPermissionProvider = Provider.family<void, BuildContext>((ref, context) {
+final checkNotificationPermissionProvider =
+    Provider.family<void, BuildContext>((ref, context) {
   Future.delayed(Duration(seconds: 4), () {
-    checkNotificationPermission().then((value) {
-      var checkPermissionStatusInLocalStorage = ref
-          .read(sharedPreferencesProvider)
-          .getBool(SharedPreferenceConstants.notificationPermission);
-      if (Platform.isAndroid &&
-          checkPermissionStatusInLocalStorage == null &&
-          value == AuthorizationStatus.denied) {
-        context.push(RouteConstants.notificationPermissionPath);
-      } else if (Platform.isIOS && value == AuthorizationStatus.notDetermined) {
-        context.push(RouteConstants.notificationPermissionPath);
-      }
-    });
+    var notNowCount = ref.read(getNotificationPermissionCountProvider);
+    if (notNowCount < 3) {
+      ref.read(notificationPermissionStatusProvider.future).then((value) {
+        var checkPermissionStatusInLocalStorage = ref
```

Comment:
Is this a boolean? Should be isPermissionStatusInLocalStorage or something like that? Using the verb "check" at the start sounds like it should be a function :)

Reply:
Thank you for letting me know. I have fixed the issue.